l1-contracts/contracts/state-transition/TestnetVerifier.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {IVerifier} from "./chain-interfaces/IVerifier.sol";
+
+contract TestnetVerifier is IVerifier {
+    IVerifier _mainVerifier;
+
+    constructor(address _verifier) {
+        _mainVerifier = IVerifier(_verifier);
+    }
+
+    function verify(
+        uint256[] calldata _publicInputs,
+        uint256[] calldata _proof,
+        uint256[] calldata _recursiveAggregationInput
+    ) external view returns (bool) {
+        // We allow skipping the zkp verification for the test(net) environment
+        // If the proof is not empty, verify it, otherwise, skip the verification
+        assert(block.chainid != 1);
```

Comment:
I would say we should now place this in the constructor. This way `TestnetVerifier` can't be deployed to mainnet at all